Define  sub dir to be used for cf push

rangulvers opened the pull request:
After your feedback I added the option to define a directory to be used. 

Use Case : if you have a collection of cf apps in one repo it is now possible to define the sub folder to be used. This way only the specific app will be pushed and no extra repos are needed. Useful for teaching purposes 

In addition a debug parameter can be set to list all subdirs. This can help to correctly setup the github action script

rangulvers committed:
Update README.md

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update action.yml

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update action.yml

rangulvers committed:
Update action.yml

rangulvers committed:
Update action.yml

rangulvers committed:
Update action.yml

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update action.yml

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Create entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update entrypoint.sh

rangulvers committed:
Update action.yml

rangulvers committed:
Update README.md

rangulvers committed:
Update README.md

rangulvers committed:
Update README.md

jhunt commented:
I'd love to merge this.  First, however, I have a few suggestions and a question.

1. Please canonicalize references to this repo (mostly in documentation)
2. Remove the notice about this being a fork (it won't be forked when it gets merged)
3. Instead of the INPUT_VALIDATE environment variable, why not just allow additional options to be set directly?  I can see five useful cases: CF_API_OPTIONS, CF_AUTH_OPTIONS, CF_TARGETOPTIONS, CF_PUSH_OPTIONS, and a global CF_OPTIONS.

Why did you drop the `set -eu` and add `-l` to the shebang?

rangulvers commented:
valid feedback. I will create a new PR and remove this one.

while testing i added the -l and removed the -eu to figure out some stuff. But never reverted back to what it should be
the INPUT_VALIDATE was coming from your origional repo. But I am already working on a version to allow for more options in the future

jhunt commented:
Thanks!

jhunt commented:
I can work up a patch for the change to option handling.
